### DISCUSSION TITLE
Access control in a field does not affect filtering by it

### DISCUSSION THREAD

**Discussion by mariomnts** [keystonejs/keystone#8240] [Questions] [answered]
I have a list which has no defined access control but one of their fields has a denied all access policy like the following but I am able to query ListKey filtered based on fieldName so I can’t really see that value but I could do brute-force and guess it. Is this something expected or this can be a security hole?

```
ListKey: list({
      fields: {
        fieldName: text({
          access: {
            read: ({ session, context, listKey, fieldKey, operation, item }) => false,
            create: ({ session, context, listKey, fieldKey, operation, inputData }) => false,
            update: ({ session, context, listKey, fieldKey, operation, inputData, item }) => false,
          },
        }),
      },
    }),
```

**Comment by borisno2** [comment, marked as answer]
Hi @mariomnts this is expected. If you want to disable filtering by a particular field you can use the `isFilterable` config option - https://keystonejs.com/docs/fields/overview#common-configuration

**Reply by dcousens** [reply]
This behavior might be too subtle, but I don't want to jump to conclusions without seeing what the canonical code could look like if documented.  I think we should add an example for nuanced access control scenarios like this.

**Reply by dcousens** [reply]
Related
- https://github.com/keystonejs/keystone/pull/8247#pullrequestreview-1249395458
